src/main/java/net/syscon/elite/service/CourtHearingsService.java

```diff
@@ -91,17 +91,36 @@ public CourtHearing scheduleHearing(final Long bookingId, final Long courtCaseId
         return courtHearing;
     }
 
-    // TODO - WIP DT-651
+    /**
+     * Returns all court hearings for a given booking ID.
+     */
     @VerifyBookingAccess
     public CourtHearings getCourtHearingsFor(final Long bookingId) {
-        return CourtHearings.builder().build();
+        final var courtHearingsBuilder = CourtHearings.builder();
+
+        courtEventRepository.findByOffenderBooking_BookingId(bookingId).stream()
```

Comment:
Note: I am not restricting to active bookings.  It could but should it?  The same goes for court cases the court hearings will be linked to.